docs/guides/es2k/build-p4-programs.md

```diff
@@ -0,0 +1,104 @@
+# Building and Deploying P4 Programs (ES2K)
+
+- [1. Overview](#1-overview)
+- [2. Install compiler and tools](#2-install-compiler-and-tools)
+- [3. Generate P4 artifacts](#3-generate-p4-artifacts)
+
+## 1. Overview
+
+This document explains how to install Intel® IPU E2100 P4 Compiler
+`p4c-pna-xxp`and Intel® Configurable Pipeline Tool `cpt` to build P4 artifacts.
+
+## 2. Install compiler and tools
+Install the `cpt`, `p4c` and `p4-sde` using prebuilt RPMs available as part of
+release package on Fedora 37 x86_64 server.
+
+  ```bash
+  # Extract RPMs from the tarball
+  tar xvzf mev-hw-b0-<release>-fedora37.tgz
+  cd host/packages/x86_64/
+
+  # Install RPMs from tarball
+  dnf localinstall cpt*.rpm --allowerasing
+  dnf localinstall p4c*.rpm --allowerasing
+  dnf localinstall p4-sde*.rpm --allowerasing
+  
+  ```
+  - Note: Default location for installed packages is under `/usr` and these rpms
+are not relocatable
+
+- After installing the RPMs, confirm the version of the cpt and p4c-pna-xxp.
+
+  ```bash
+  # Set environment variables
+  export LD_LIBRARY_PATH=/usr/lib:/usr/lib64:/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
+  
+  [root@host ~]# cpt --version
+  Intel(R) Configurable Pipeline Tool Version 3.8.0.16
+  Copyright (C) 2022 Intel Corporation.  All rights reserved.
+
+  [root@host ~] # p4c-pna-xxp --version
+  p4c-pna-xxp
+  Version 3.0.70.7
+  ```
+
+  Note: If you see boost lib version mismatch while running p4c-pna-xxp, install
+that boost from source tar ball to fix the problem.
+  ```bash
+  p4c-pna-xxp --version
+  p4c-pna-xxp: error while loading shared libraries: \
+  libboost_iostreams   -.so.1.69.0: cannot open shared object file \:
+  No such file or directory
+  ```
+
+  - Before compiling boost_1_69_0, ensure Development Tools are installed.
+   ```bash
+   wget https://boostorg.jfrog.io/artifactory/main/release/1.69.0/source/boost_1_69_0.tar.gz
+   tar -xf boost_1_69_0.tar.gz
+   cd boost_1_69_0/
+   ./bootstrap.sh
+   ./b2
+   ./b2  install
+   ```
+
+## 3. Generate P4 artifacts
+Use the `p4c-pna-xxp` and `cpt` tools installed to create artifacts.
+See below example for more details for simple_l3_l4_pna.p4.
```

Comment:
What we want to do now is:

1. Explain how to generate artifacts for a sample program (simple_l3_l4_pna.p4).
2. Tell the reader where to find other sample programs.

We want to break the sequence into digestible chunks, making each of those chunks a named subsection.

3.1 Generate artifacts
3.2 Generate .pkg file
3.3 P4 reference files

First, delete line 66. We will replace it with the beginning of section 3.1.